src/gpu/generic/sycl/reduction_kernels.hpp

```diff
@@ -0,0 +1,659 @@
+/*******************************************************************************
+* Copyright 2023-2024 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_GENERIC_SYCL_REDUCTION_KERNELS_HPP
+#define GPU_GENERIC_SYCL_REDUCTION_KERNELS_HPP
+
+#include "common/c_types_map.hpp"
+#include "common/dnnl_thread.hpp"
+#include "common/primitive_exec_types.hpp"
+#include "common/utils.hpp"
+#include "gpu/generic/sycl/sycl_io_helper.hpp"
+#include "gpu/generic/sycl/sycl_math_utils.hpp"
+#include "gpu/generic/sycl/sycl_primitive_conf.hpp"
+#include "xpu/sycl/memory_storage_base.hpp"
+#include "xpu/sycl/types.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace generic {
+namespace sycl {
+
+struct init_kernel_t {
+    xpu::sycl::out_memory_arg_t out_arg_;
+    dnnl_alg_kind_t alg_;
```

Comment:
Please replace with `alg_kind_t`. Please update all algorithms with `alg_kind::name`. They are meant to be used from an `alg_kind` namespace, not the public API C type.

Reply:
This is done. I have applied similar changes to avoid using `dnnl_data_type_t` and `dnnl_dim_t`.